### PR TITLE
Serve robots.txt for Prow

### PR DESCRIPTION
**Pull request by ammarlakis**
**Description**

Changes proposed in this pull request:

- New deployment of an Nginx reverse proxy that can be configured through a ConfigMap. One caveat is that we need a deployment reloader to restart nginx when the ConfigMap changes.

Other approaches considered:
- Using GCE ingress that would point to the bucket backend where the robots.txt file is hosted; This is not supported yet  https://github.com/kubernetes/ingress-gce/issues/33
- Using Nginx as an ingress; changing the existing ingress would mean losing the features and integration it provides.

**Related issue(s)**
- Fixes https://github.com/kyma-project/website/issues/576

### CODE REVIEW

File: prow/README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@ Its structure looks as follows:
   └── plugins.yaml          # The file with Prow plugins configuration
   └── create-secrets-for-workload-cluster.sh  # The script which creates a Secret in the Prow cluster, used to access the workload cluster
   └── set-up-workload-cluster.sh  # The script which prepares the workload cluster to be used by the Prow cluster
+  └── static-files          # Files that will be uploaded to the nginx web server. To configure how traffic is routed to these files, update the prow ingress paths and the web server configuration ConfigMap.
```
Comment (dekiel) [inline prow/README.md:55]: This is not the right place to provide details how to work with this files. Please update prow/cluster/README.md with such details. Place some info about how content of static-files is updated.

**Review by dekiel — CHANGES_REQUESTED** [review]
Please extend operational documentation how to work with this component.

File: prow/README.md (hunk 1 above)
Comment (ammarlakis) [reply prow/README.md:55]: @dekiel the folder is at `prow/static-files` not `prow/cluster/static-files`

Comment (dekiel) [reply prow/README.md:55]: I know, but still you put information about how to work with files in place where we have short description what a directory contains. prow/cluster/README.md is my proposal where to place some details about how to use and work with these files. I would even vote for moving static-files somewhere under prow/cluster directory as these files are used on a prow cluster only. They are not a standalone tool and have not more usage than prow cluster.

File: prow/cluster/README.md [hunk 2]
```diff
@@ -11,6 +11,10 @@ This folder contains files related to the configuration of the Prow production c
...
 ```
-  ├── components                        # Definitions of Prow components and cluster configuration.
-  └── resources                         # Helm charts used by the Prow cluster.
+  ├── components            # Definitions of Prow components and cluster configuration.
+  ├── resources             # Helm charts used by the Prow cluster.
+  └── static-files          # Files that will be uploaded to the nginx web server.
 ```
+
+##  Adding static files
+All files added to the `static-files` folder are uploaded in a ConfigMap to the cluster and mounted by the web server in the web root directory. To route traffic for a specific path to the nginx web server, update the ingress `tls-ing` configuration in `tls-ing_ingress.yaml`.
```
Comment (dekiel) [inline prow/cluster/README.md:20]:
```suggestion
All files added to the `static-files` folder are automatically uploaded by Prow config_updater plugin to the cluster in a ConfigMap. Uploaded files are mounted by the web server in the web root directory. To route traffic for a specific path to the nginx web server, in order to serve these files, update the ingress `tls-ing` configuration in `tls-ing_ingress.yaml`.
```

**Review by dekiel — APPROVED** [review]

File: prow/cluster/README.md [hunk 3]
```diff
@@ -11,6 +11,10 @@ This folder contains files related to the configuration of the Prow production c
...
 ```
-  ├── components                        # Definitions of Prow components and cluster configuration.
-  └── resources                         # Helm charts used by the Prow cluster.
+  ├── components            # Definitions of Prow components and cluster configuration.
+  ├── resources             # Helm charts used by the Prow cluster.
+  └── static-files          # Files that will be uploaded to the nginx web server.
 ```
+
+##  Adding static files
+All files added to the `static-files` folder are automatically uploaded by Prow config_updater plugin to the cluster in a ConfigMap. Uploaded files are mounted by the web server in the web root directory. To route traffic for a specific path to the nginx web server, in order to serve these files, update the ingress `tls-ing` configuration in `tls-ing_ingress.yaml`.
```
Comment (alexandra-simeonova) [inline prow/cluster/README.md:20]:
```suggestion
###  Adding static files
All files added to the `static-files` folder are automatically uploaded by the Prow `config_updater` plugin to the cluster in a ConfigMap. Uploaded files are mounted by the web server in the web root directory. To route traffic for a specific path to the NGINX web server, in order to serve these files, update the Ingress `tls-ing` configuration in `tls-ing_ingress.yaml`.
```

**Review by dekiel — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**